File: leetcode/1673-find-the-most-competitive-subsequence/mostCompetitive.test.ts
import cases from "./cases.json";
import mostCompetitive from "./mostCompetitive";

test("find the most competitive subsequence", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = mostCompetitive(testCase.input.nums, testCase.input.k);
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});

File: leetcode/1673-find-the-most-competitive-subsequence/mostCompetitive_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"reflect"
	"testing"

	"github.com/edram/leetcode/util"
)

type MostCompetitiveCase struct {
	Input struct {
		Nums []int
		K    int
	}

	Output []int
}

func TestMostCompetitive(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]MostCompetitiveCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := mostCompetitive(testCase.Input.Nums, testCase.Input.K)
		if !reflect.DeepEqual(result, testCase.Output) {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}
